### PR TITLE
(Chore) Remove authenticated selector

### PR DESCRIPTION
**Pull request by janjaap**
This PR contains changes that prevents component from rerendering when the `makeSelectIsAuthenticated` selector updates its value and passes a different prop value to the appropriate components. 
The previous asynchronous behaviour caused several issues.; logging in wouldn't always redirect the user to the incidents overview page and the header component would render its `tall` version and, after getting a different value for the authentication, would render its `short` version.

### CODE REVIEW

File: src/containers/App/reducer.js [hunk 1]
```diff
@@ -29,16 +33,28 @@ export const initialState = fromJS({
...
     mainToSub: {},
-  }
+  },
 });
 
 function appReducer(state = initialState, action) {
   switch (action.type) {
+    case AUTHENTICATE_USER:
+      if (action.payload && action.payload.accessToken) {
+        global.sessionStorage.setItem(ACCESS_TOKEN, action.payload.accessToken);
```
Comment (jasperswart) [inline src/containers/App/reducer.js]: the global.sessionStorage is being set in the saga already: the login and logout functions in auth.js

File: src/containers/App/reducer.js [hunk 2]
```diff
@@ -58,27 +74,38 @@ function appReducer(state = initialState, action) {
...
+          progress: action.payload,
+        }),
+      );
 
     case UPLOAD_SUCCESS:
     case UPLOAD_FAILURE:
+      return state.set('upload', fromJS({}));
+
+    case LOGOUT:
+      global.sessionStorage.removeItem(ACCESS_TOKEN);
```
Comment (jasperswart) [inline src/containers/App/reducer.js]: de global.sessionStorage is being set in the saga already: the login and logout functions in auth.js

File: src/shared/services/api/api.test.js [hunk 3]
```diff
@@ -17,17 +16,46 @@ describe('api service', () => {
   const queryString = 'name1=value1&name2=value2&value3=foo&value3=bar';
   const url = 'https://url/to/test';
   const token = 'bearer-token';
+  let origSessionStorage;
```
Comment (jasperswart) [inline src/shared/services/api/api.test.js]: why do you not use `getAccessToken.mockImplementation(() => 'mock')` here?

File: src/signals/incident/containers/IncidentContainer/index.test.js [hunk 4]
```diff
@@ -5,24 +5,38 @@ import { withAppContext } from 'test/utils';
...
     props = {
       incidentContainer: { incident: {} },
       getClassification: jest.fn(),
       updateIncident: jest.fn(),
       createIncident: jest.fn(),
-      isAuthenticated: false
+    };
+    origSessionStorage = global.sessionStorage;
+
+    global.sessionStorage = {
```
Comment (jasperswart) [inline src/signals/incident/containers/IncidentContainer/index.test.js]: why do you not use `getAccessToken.mockImplementation(() => 'mock')` here to mock api service?

File: src/containers/App/reducer.js (hunk 1 above)
Comment (janjaap) [reply src/containers/App/reducer.js]: Good call. Removed it.

File: src/containers/App/reducer.js (hunk 2 above)
Comment (janjaap) [reply src/containers/App/reducer.js]: Another good call; also removed it.

File: src/shared/services/api/api.test.js (hunk 3 above)
Comment (janjaap) [reply src/shared/services/api/api.test.js]: I've added a package that mocks the sessionstorage instead.

File: src/signals/incident/containers/IncidentContainer/index.test.js (hunk 4 above)
Comment (janjaap) [reply src/signals/incident/containers/IncidentContainer/index.test.js]: See previous comment

File: src/shared/services/api/api.test.js (hunk 3 above)
Comment (jasperswart) [reply src/shared/services/api/api.test.js]: die package hoef je niet te gebruiken, als je:

import { getAccessToken } from 'shared/services/auth/auth';
jest.mock('shared/services/auth/auth');

en dan in elke test: 
getAccessToken.mockImplementation(() => true) // of false

dan laad hij geen andere services in.

dat zit ook op andere plekken die van getAccessToken gebruikmaken.

Comment (janjaap) [reply src/shared/services/api/api.test.js]: Problem is that, when I start mocking that module, I need to make more changes in other, unrelated, components. Even if I restore mocks after the test suite has run. Using the localstorage mock package overcomes that problem.

File: src/shared/services/api/api.test.js [hunk 5]
```diff
@@ -1,6 +1,5 @@
-import { call, select } from 'redux-saga/effects';
+import { call } from 'redux-saga/effects';
 import request from 'utils/request';
-import { makeSelectAccessToken } from 'containers/App/selectors';
 
 import { generateParams, authCall, authCallWithPayload, authPostCall, authPatchCall } from './api';
```
Comment (jasperswart) [inline src/shared/services/api/api.test.js]: the jest.mock('containers/App/selectors' can be removed

File: src/shared/services/api/api.test.js (hunk 3 above)
Comment (jasperswart) [reply src/shared/services/api/api.test.js]: ok laat maar dan

File: src/shared/services/api/api.test.js (hunk 5 above)
Comment (janjaap) [reply src/shared/services/api/api.test.js]: You're right; I've removed it.

**Pull request merged**